content/guides/quickstart/groups-guide.md

```diff
@@ -0,0 +1,1535 @@
+    =.  gids  (skim gids |=(=gid =(our.bol host.gid)))
+    ``noun+!>(`(set gid)`(~(gas in *(set gid)) gids))
+  ::
+      [%x %squad @ @ ~]
+    =/  =gid  [(slav %p i.t.t.path) i.t.t.t.path]
+    ``noun+!>(`(unit squad)`(~(get by squads) gid))
+  ::
+      [%x %acl @ @ ~]
+    =/  =gid  [(slav %p i.t.t.path) i.t.t.t.path]
+    =/  u-squad=(unit squad)  (~(get by squads) gid)
+    :^  ~  ~  %noun
+    !>  ^-  (unit [pub=? acl=ppl])
+    ?~  u-squad
+      ~
+    `[pub.u.u-squad (~(get ju acls) gid)]
+  ::
+      [%x %members @ @ ~]
+    =/  =gid  [(slav %p i.t.t.path) i.t.t.t.path]
+    ``noun+!>(`ppl`(~(get ju members) gid))
+  ::
+      [%x %titles ~]
+    :^  ~  ~  %json
+    !>  ^-  json
+    :-  %a
+    %+  turn
+      (sort ~(tap by squads) |=([[* a=@t *] [* b=@t *]] (aor a b)))
+    |=  [=gid =@t ?]
+    ^-  json
+    %-  pairs:enjs:format
+    :~  :-  'gid'
+        %-  pairs:enjs:format
+        :~  ['host' s+(scot %p host.gid)]
+            ['name' s+name.gid]
+        ==
+        ['title' s+t]
+    ==
+  ==
+::
+++  on-arvo
+  |=  [=wire =sign-arvo]
+  ^-  (quip card _this)
+  ?.  ?=([%bind ~] wire)
+    (on-arvo:def [wire sign-arvo])
+  ?.  ?=([%eyre %bound *] sign-arvo)
+    (on-arvo:def [wire sign-arvo])
+  ~?  !accepted.sign-arvo
+    %eyre-rejected-squad-binding
+  `this
+::
+++  on-fail  on-fail:def
+--
+```
+
+### Marks
+
+Marks are Urbit's version of filetypes/MIME types (but strongly typed and with
+inter-mark conversion methods). We need to define a mark for the `act`ions we'll
+send or receive, and the `upd`ates we'll send to subscribers or receive for
+subscriptions. These will be very simple since we don't need to do any
+conversions to things like JSON.
+
+Mark files are stored in the `/mar` directory of a desk. Save the
+`%squad-do` mark in `squad/mar/squad/do.hoon`, and the `%squad-did`
+mark in `squad/mar/squad/did.hoon`.
+
+#### `%squad-do`
+
+```hoon {% copy=true %}
+/-  *squad
+|_  a=act
+++  grow
+  |%
+  ++  noun  a
+  --
+++  grab
+  |%
+  ++  noun  act
+  --
+++  grad  %noun
+--
+```
+
+#### `%squad-did`
+
+```hoon {% copy=true %}
+/-  *squad
+|_  u=upd
+++  grow
+  |%
+  ++  noun  u
+  --
+++  grab
+  |%
+  ++  noun  upd
+  --
+++  grad  %noun
+--
+```
+### Front-end
+
+We could have put the front-end code directly in our Gall agent, but it tends to
+be quite large so it's convenient to have it in a separate file and just import
+it. Most of this file consists of Sail code, which is the internal HTML
+representation, similar to other server-side renderings like Clojure's Hiccup.
+
+Save the code below in `squad/app/squad/index.hoon`.
+
+```hoon {% copy=true mode="collapse" %}
+/-  *tally, *squad
+|=  [bol=bowl:gall =by-group voted=(set pid) withdrawn=(set pid)]
+^-  manx
+?.  .^(? %gu /(scot %p our.bol)/squad/(scot %da now.bol))
+  ;html
+    ;head
+      ;title: Tally
+      ;meta(charset "utf-8");
+      ;style
+        ;+  ;/
+            ^~
+            ^-  tape
+            %-  trip
```

Comment:
is there a reason to prefer tripping a cord rather than just having a tape here?

Reply:
yes it's cos css is full of curly braces which need escaping. I just realised I pasted in the wrong code here though lol, this is the front-end for tally not squad. I will fix